thunder/core/transforms.py

```diff
@@ -3466,7 +3472,8 @@ def put_grad(v: Variable, val: Any) -> None:
         # Having a single cotangent is a common case, so we flatten it
         # Otherwise, we will need to rewrite the pullback functions
         cotangents = tree_flatten(cotangents)[0]
-        residuals = forward_env[symbol_output[0].name].residuals
+        residuals = [] if symbol_output[0] is None else forward_env[symbol_output[0].name].residuals
+        # residuals = forward_env[symbol_output[0].name].residuals
```

Comment:
Revert this change, please. Equivalent options are:
* Use `if symbol.output is None: continue` in the beginning of the loop
* Use `for symbol in reversed(list(iter_bound_symbols(trace.bound_symbols))):`